extensions/blocks/pinterest/pinterest.php

```diff
@@ -25,6 +29,141 @@ function register_block() {
+		$placeholder = sprintf(
+			// The AMP_Img_Sanitizer will convert his to <amp-img> while also supplying `noscript > img` as fallback when JS is disabled.
+			'<a href="%s" placeholder><img src="%s" alt="%s" layout="fill" object-fit="contain" object-position="top left"></a>',
+			esc_url( $attr['url'] ),
+			esc_url( $image['url'] ),
+			esc_attr( $title )
+		);
+
+		$amp_padding     = 5;   // See <https://github.com/ampproject/amphtml/blob/b5dea36e0b8bd012585d50839766a084f99a3685/extensions/amp-pinterest/0.1/amp-pinterest.css#L269>.
+		$amp_fixed_width = 237; // See <https://github.com/ampproject/amphtml/blob/b5dea36e0b8bd012585d50839766a084f99a3685/extensions/amp-pinterest/0.1/amp-pinterest.css#L270>.
+		$pin_info_height = 60;  // Minimum Obtained by measuring the height of the .-amp-pinterest-embed-pin-text element.
+
+		// Add height based on how much description there is. There are roughly 30 characters on a line of description text.
+		$has_description = false;
+		if ( ! empty( $info['description'] ) ) {
+			$desc_padding_top = 5;  // See <https://github.com/ampproject/amphtml/blob/b5dea36e0b8bd012585d50839766a084f99a3685/extensions/amp-pinterest/0.1/amp-pinterest.css#L342>.
+			$pin_info_height += $desc_padding_top;
+
+			// Trim whitespace on description if there is any left, use to calculate the likely rows of text.
+			$description = trim( $info['description'] );
+			if ( strlen( $description ) > 0 ) {
+				$has_description  = true;
+				$desc_line_height = 17; // See <https://github.com/ampproject/amphtml/blob/b5dea36e0b8bd012585d50839766a084f99a3685/extensions/amp-pinterest/0.1/amp-pinterest.css#L341>.
+				$pin_info_height += ceil( strlen( $description ) / 30 ) * $desc_line_height;
+			}
+		}
+
+		if ( ! empty( $info['repin_count'] ) ) {
+			$pin_stats_height = 16;  // See <https://github.com/ampproject/amphtml/blob/b5dea36e0b8bd012585d50839766a084f99a3685/extensions/amp-pinterest/0.1/amp-pinterest.css#L322>.
+			$pin_info_height += $pin_stats_height;
+		}
+
+		// When Pin description is empty, make sure title and description from rich metadata are supplied for accessibility and discoverability.
+		$title = $has_description ? '' : implode( "\n", array_filter( array( $title, $description ) ) );
+
+		$amp_pinterest = sprintf(
+			'<amp-pinterest style="%1$s" data-do="embedPin" data-url="%2$s" width="%3$d" height="%4$d" title="%5$s">%6$s</amp-pinterest>',
+			esc_attr( 'line-height:1.5; font-size:21px' ), // Override styles from theme due to precise height calculations above.
+			esc_url( $attr['url'] ),
+			$amp_fixed_width + ( $amp_padding * 2 ),
+			$image['height'] + $pin_info_height + ( $amp_padding * 2 ),
+			esc_attr( $title ),
+			$placeholder
+		);
+	} else {
+		// Fallback embed when info is not available.
+		$amp_pinterest = sprintf(
+			'<amp-pinterest data-do="embedPin" data-url="%s" width="%d" height="%d"></amp-pinterest></div>',
+			esc_url( $attr['url'] ),
+			450, // Fallback width.
+			750 // Fallback height.
+		);
```

Comment:
If we do not have info about the pin (like when embedding a user profile such as `https://pinterest.com/anapinskywalker/`), is it worth using the `amp-pinterest` tag? Wouldn't we be better off just outputting a link?

Reply:
I'm trying to embed `https://pinterest.com/anapinskywalker/` in the Pinterest block but it's not doing anything. Is that expected?

![image](https://user-images.githubusercontent.com/134745/92407775-67a67180-f0f0-11ea-901f-78952e4c5c33.png)

Clicking Embed does nothing. So a user wouldn't be able to embed non-pins anyway, right?

But yes, it would be better to show a link if a non-pin ended up getting embedded.

Otherwise, if the only issue is the info request fails, then it still seems a bit better to render `amp-pinterest` rather than just a bare URL link. While the size of the Pin may be not ideal, at least it should be showing the Pin.

I'm also going to open an issue in the amphtml repo to request that the `width`/`height` of `amp-pinterest` be optional in the same way as they are for `amp-twitter` or `amp-gist`. It is too tedious to get the dimensions exactly right. So once the embed initializes, it should resize itself to fit its contents even if this causes some CLS. This is an instance of where AMP needs to be pragmatic.

Reply:
> So once the embed initializes, it should resize itself to fit its contents even if this causes some CLS. This is an instance of where AMP needs to be pragmatic.

See https://github.com/ampproject/amphtml/issues/10318#issuecomment-688501998.

Reply:
> If we do not have info about the pin (like when embedding a user profile such as `https://pinterest.com/anapinskywalker/`), is it worth using the `amp-pinterest` tag? Wouldn't we be better off just outputting a link?

I've added a link to render as a fallback in this case: b9e4f62345c9dbaa599e2b0f2427cac711aa1fa2.

Reply:
> I'm trying to embed https://pinterest.com/anapinskywalker/ in the Pinterest block but it's not doing anything. Is that expected?

That's odd. It seems to work well on my end:

![image](https://user-images.githubusercontent.com/426388/92451218-bcafb980-f1bc-11ea-8a90-cb969c106dba.png)

This is a test account set up by Pinterest themselves, but you could try with just about any other profile:

![image](https://user-images.githubusercontent.com/426388/92451333-dcdf7880-f1bc-11ea-94f9-01d0a59e2a06.png)